project_wip_material/consumption_route.py

```diff
@@ -141,21 +152,32 @@ def _create_or_update_consumption_picking_types(self):
             self._create_consumption_picking_types()
 
     def _get_consumption_pull_values(self):
+        source_location = (
+            self.lot_stock_id if _has_one_step_consumption(self) else self.consu_prep_location_id
+        )
+        procure_method = 'make_to_stock' if _has_one_step_consumption(self) else 'make_to_order'
         return {
-            'name': self._format_rulename(self.lot_stock_id, self.consu_location_id, 'Production'),
-            'location_src_id': self.lot_stock_id.id,
+            'name': self._format_rulename(source_location, self.consu_location_id, ''),
+            'location_src_id': source_location.id,
             'location_id': self.consu_location_id.id,
             'picking_type_id': self.consu_type_id.id,
             'action': 'move',
             'active': True,
             'company_id': self.company_id.id,
+            'sequence': 1,
+            'propagate': True,
+            'procure_method': procure_method,
+            'group_propagation_option': 'propagate',
         }
 
+    def _get_consumption_route_description(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4160521)